driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -237,6 +238,7 @@ private Metrics metrics() {
      */
     class SpeculativeExecution implements Connection.ResponseCallback {
         final String id;
+        private final Message.Request request;
```

Comment:
Not much difference really, if anything it makes it clear that the call to `callback.request()` always returns the same object. I'll leave it this way for now since it's not part of the fix.